pages/02.Tutorials/11.object-storage-acls/docs.en.md

```diff
@@ -0,0 +1,166 @@
+
+```shell
+# .s3cfg-admin represents your personal s3cmd config containing your access/secret key pair.
+s3cmd -c .s3cfg-admin mb s3://acl-read
+s3cmd -c .s3cfg-admin mb s3://acl-write
+# Create a test.txt file
+echo "secret-text" > test.txt
+# Upload the file in both buckets
+s3cmd -c .s3cfg-admin put test.txt s3://acl-read/test.txt
+s3cmd -c .s3cfg-admin put test.txt s3://acl-write/test.txt
+# We will later use the test.txt object to confirm our applied ACLs are working
+```
+
+As we did not define any ACLs while creating the buckets/objects, the default private ACL will be used. In the current state all users, who have access to our OpenStack project (users who have created EC2 credentials for this project), have full control on our newly created buckets/objects.
+
+For an object to be accessable to an user, the object itself and the parent buckets have to have proper ACLs set.
+
+### Revoke default ACLs 
+
+In this step we will revoke the access to the buckets/objects for every user with access to our OpenStack project.
+
+*As this point it is important to understand that OpenStack EC2 credentials are bound to a user and a project. Thus if you are managing multiple OpenStack projects, there is more then 1 way to allow an user to be able to access your buckets/objects. You may allow the users EC2 credentials created for project A to access your data but at the same time revoke the access for the users EC2 credentials created for project B* 
+
+```shell
+# Get our OpenStack project ID
+MY_PROJECT_ID=$(openstack project list -f value -c ID)
+# Revoke access to anyone except ourself from your OpenStack project
+s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-read
+s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-write
+# At this point already the objects will only be accessable by the owner. To make it more clean we will also revoke the default grant of the objects themself.
+s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-read/test.txt
+s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-write/test.txt
+```
+
+Now only the owner (.s3cfg-admin) has access to the buckets and objects we created for this tutorial. 
+
+### Grant read-only access via user ACLs
+
+We will proceed to grant read rights for our additional user.
+
+```shell
+# Assumption : user max.mustermann@myproject.de has access to our project
+# Lets grant him read access to your acl-read bucket and all objects contained inside (using the recursive option)
+s3cmd -c .s3cfg-admin setacl --acl-grant=read:u:max.mustermann@myproject.de/${MY_PROJECT_ID} s3://acl-read --recursive
+# We may also only grant the user to specific objects inside the bucket, in this case we leave out the recursive option and directly add the object names :
```

Comment:
strange, I get an error here:
```
$ s3cmd  setacl --acl-grant=read:u:${OS_USERNAME}/${MY_PROJECT_ID} s3://mytestbucket-read --recursive
ERROR: S3 error: 400 (UnresolvableGrantByEmailAddress): The e-mail address you provided does not match any account on record.
```
First I had the same error with an extra user that I created, so I spent a lot of time searching if I set up the user wrong. But then I tried it with my normal user... and it still failed.

Reply:
It worked if I used the openstack user ID instead of the email address... but then `s3cmd info` showed as ACLs
```
   ACL:       ${MY_PROJECT_ID}: FULL_CONTROL
   ACL:       ${MY_PROJECT_ID}: READ
```

Reply:
I adjusted the documentation accordingly. I tested with usernames before but assumed it would make no difference using mail addresses. Now I could confirm the ACL cannot be set for users with mail addresses as names. I will communicate the issue together with the other ACL related problems to Quobyte.